lib/core/include/irods/irods_server_properties.hpp

```diff
@@ -142,12 +154,26 @@ namespace irods
         }
 
     private:
+        /// @brief Acquire a read lock, that is, a lock that is not exclusive, that is why it returns a shared_lock
+        /// here.
+        std::shared_lock<std::shared_mutex> acquire_read_lock() const
+        {
+            return std::shared_lock<std::shared_mutex>(property_mutex_);
+        }
+
+        /// @brief Acquire an exclusive lock for writing(Hence the unique_lock)
```

Comment:
Please elaborate more on the reason why `std::unique_lock` is required for correctness.

It is important for the next reader.